### ISSUE TITLE
How to run on multiple directories

### ISSUE THREAD

**Issue by sergevonage** [golangci/golangci-lint-action#271]
I have a monorepo and want run linter on `dir1` and `dir2`. How can I do it?
```
working-directory: dir1 dir2
```
doesn't work:
```
working-directory (dir1  dir2) was not a path
```

**Comment by ChrisSargent** [comment]
I also have this question.

We have a mixed codebase like:

```
packages
- Go
- -- cmd
- ---- module1
- ---- module2
- Typescript
```

Currently, we run linting locally with a command like this from the 'Go' directory: `find . -type f -name go.mod -maxdepth 3 -execdir golangci-lint run --fast \;`

I have tried a few times to set it up in a better way but I always end up hitting the 'there are no go files error'.

Is there a way to install this action (so we get all the right installations and caches) but use a custom command to run?

**Comment by jonny-rimek** [comment]
i was able to lint multiple directories by simply adding all the directories `golint -set_exit_status cmd/... internal/... ui/...` important note, is to add /... so that all subdirectories are linted as well.

maybe they added the feature in the meantime

**Comment by KEINOS** [comment]
Finally, I found out how.

## tl; dr

You need to place them in the `args` directive.

## ts; dr

```sh
golangci-lint run --config ./.github/golangci.yml ./... ./_example/dir1 ./_alt/dir2
```

The above case will be:

```yaml
name: golangci-lint

on:
  workflow_dispatch:
  pull_request:
    branches:
      - main

jobs:
  golangci:
    name: lint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: golangci-lint
        uses: golangci/golangci-lint-action@v3
        with:
          version: latest
          args: --config ./.github/golangci.yml ./... ./_example/dir1 ./_alt/dir2
```

```yaml

```

**Comment by serbrech** [comment]
@KEINOS I thought that the `./...` would already run it on `./_example/dir` and on `./_alt/dir2` since it runs it recursively on all folder below `.`

**Comment by KEINOS** [comment]
@serbrech

I thought so too. But since the directory begins with "_", it omits from "./..." unless specifying them. It took me time to find it out.

**Comment by ChrisSargent** [comment]
FYI - just ran in to this issue again. Easiest workaround we found was to pass `--help` as an arg, which prevents the linters from actually running in this action. Then we can use our own custom commands for running it later in the job.

**Comment by faddat** [comment]
this is on topic as it gets (daily user)

**Comment by faddat** [comment]
@serbrech -- but it does not :D

**Comment by brandur** [comment]
We needed to be able to do this on multiple Go modules that we have in the same repository. The trick of passing `args: <dir1> <dir> <dir3>` didn't work because golangci-lint won't do more than one module in a pass.

Chris' answer above to configure golangci-lint's GitHub Action with `args: --help` to trick it into downloading the binary, not actually run, then adding a separate step to do the linting, seems to have done the trick. Full code:

``` yaml
      - name: Lint
        uses: golangci/golangci-lint-action@v4
        with:
          # golangci-lint needs to be run separately for every Go module, and
          # its GitHub Action doesn't provide any way to do that. Have it fetch
          # the golangci-lint binary, trick it into not running by sending only
          # `--help`, then run the full set of lints below. DO NOT run separate
          # modules as separate golangci-lint-action steps. Its post run caching
          # can be extremely slow, and that's amplified in a very painful way if
          # it needs to be run multiple times.
          args: --help
          version: ${{ env.GOLANGCI_LINT_VERSION }}

      - name: Run lint
        run: make lint
```

With our make target being:

``` make
.PHONY: lint
lint:
	cd . && golangci-lint run --fix
	cd cmd/river && golangci-lint run --fix
	cd riverdriver && golangci-lint run --fix
	cd riverdriver/riverdatabasesql && golangci-lint run --fix
	cd riverdriver/riverpgxv5 && golangci-lint run --fix
	cd rivertype && golangci-lint run --fix
```

A side effect is that it no longer annotates the GitHub PR with lint failures like the GitHub Action step would (`--out-format=github-actions`), but if brings filenames and line numbers back, personally I don't think that's a downside.

**Comment by ldez** [comment]
duplicate of #74
